File: backend/auth-node/src/controllers/floorPlanController.js
import { FloorApiClient } from '../services/floorApiClient.js';

const floorApiClient = new FloorApiClient();

export class FloorPlanController {
  static async checkHealth(req, res) {
    try {
      const healthy = await floorApiClient.isHealthy();
      
      if (healthy) {
        return res.status(200).json({});
      } else {
        return res.status(503).json({});
      }
    } catch (error) {
      return res.status(503).json({});
    }
  }

  static async analyzeFloorPlan(req, res) {
    try {
      if (!req.file) {
        return res.status(400).json({
          status: 'error',
          message: 'Файл не может быть пустым'
        });
      }

      const contentType = req.file.mimetype;
      if (!contentType || !contentType.startsWith('image/')) {
        return res.status(400).json({
          status: 'error',
          message: 'Файл должен быть изображением (JPG, PNG)'
        });
      }

      console.log('Получен запрос на анализ планировки:', req.file.originalname);

      const response = await floorApiClient.analyzeFloorPlan(req.file);

      console.log('Анализ завершен. Комнат:', response.statistics?.rooms,
        ', Стен:', response.statistics?.walls,
        ', Углов:', response.statistics?.corners);

      return res.status(200).json({
        status: response.status,
        blueprint3d: response.blueprint3d,
        statistics: response.statistics
      });
    } catch (error) {
      console.error('Ошибка при анализе:', error);
      return res.status(500).json({
        status: 'error',
        message: 'Ошибка при анализе планировки: ' + error.message
      });
    }
  }
}

File: $RECYCLE.BIN/S-1-5-21-468433717-555027351-2740982991-1002/$RQXU4IE/src/main/java/com/_x/TestRegister1/Controller/FloorPlanController.java
package com._x.TestRegister1.Controller;

import com._x.TestRegister1.Service.FloorApiClient;
import com._x.TestRegister1.DTO.FloorPlanAnalysisResponse;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;
import org.springframework.web.multipart.MultipartFile;

import java.util.HashMap;
import java.util.Map;

@RestController
@RequestMapping("/api/floorplan")
public class FloorPlanController {
    private final FloorApiClient floorApiClient;
    private static final Logger log= LoggerFactory.getLogger(FloorPlanController.class);


    public FloorPlanController(FloorApiClient floorApiClient) {
        this.floorApiClient = floorApiClient;
    }

    @GetMapping("/health")
    public ResponseEntity<Map<String,String>> checkHealth(){
        Map<String,String> response=new HashMap<>();
        boolean healthy=floorApiClient.isHealthy();

        if (healthy) {
            return ResponseEntity.ok(response);
        } else {
            return ResponseEntity.status(HttpStatus.SERVICE_UNAVAILABLE)
                    .body(response);
        }

    }
    @PostMapping("/analyze")
    public ResponseEntity<Map<String,Object>> analyzeFloorPlan(
            @RequestParam("file") MultipartFile file
    ) {
        log.info("Получен запрос на анализ планировки: {}", file.getOriginalFilename());

        Map<String, Object> errorResponse = new HashMap<>();
        if (file.isEmpty()) {
            errorResponse.put("status", "error");
            errorResponse.put("message", "Файл не может быть пустым");
            return ResponseEntity.badRequest()
                    .body(errorResponse);
        }

        String contentType = file.getContentType();
        if (contentType == null || !contentType.startsWith("image/")) {
            errorResponse.put("status", "error");
            errorResponse.put("message", "Файл должен быть изображением (JPG, PNG)");
            return ResponseEntity.badRequest()
                    .body(errorResponse);
        }

        try {
            FloorPlanAnalysisResponse response = floorApiClient.analyzeFloorPlan(file);

            log.info("Анализ завершен. Комнат: {}, Стен: {}, Углов: {}",
                    response.getStatistics().getRooms(),
                    response.getStatistics().getWalls(),
                    response.getStatistics().getCorners());

            Map<String, Object> result = new HashMap<>();
            result.put("status", response.getStatus());
            result.put("blueprint3d", response.getBlueprint3d());
            result.put("statistics", response.getStatistics());

            return ResponseEntity.ok(result);

        } catch (Exception e) {
            log.error("Ошибка при анализе: {}", e.getMessage(), e);
            errorResponse.put("status", "error");
            errorResponse.put("message", "Ошибка при анализе планировки: " + e.getMessage());
            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR)
                    .body(errorResponse);
        }
    }
    }

